File: Ejercicios de Deber/Ejercicio 2/C++/GestionProductos.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    // Abrimos el archivo
    ifstream archivo("productos.txt");
    
    // Verificamos si se pudo abrir
    if (!archivo.is_open()) {
        cout << "Error: No se pudo abrir el archivo productos.txt" << endl;
        return 1;
    }

    string nombre, productoCaro;
    double precio, totalGeneral = 0, precioMaximo = -1;
    int cantidad;

    // Encabezados simples
    cout << "PRODUCTO  PRECIO  CANT.  SUBTOTAL" << endl;
    cout << "------------------------------------" << endl;

    // Lectura palabra por palabra: nombre (string), precio (double), cantidad (int)
    while (archivo >> nombre >> precio >> cantidad) {
        double subtotal = precio * cantidad;
        totalGeneral += subtotal;

        // Lógica para el producto mayor
        if (precio > precioMaximo) {
            precioMaximo = precio;
            productoCaro = nombre;
        }

        // Salida básica usando espacios o tabulaciones (\t)
        cout << nombre << "  " << precio << "  " << cantidad << "  " << subtotal << endl;
    }

    cout << "------------------------------------" << endl;
    cout << "TOTAL GENERAL: " << totalGeneral << endl;
    cout << "PRODUCTO MAS CARO: " << productoCaro << " (" << precioMaximo << ")" << endl;

    archivo.close();
    return 0;
}

File: Ejercicios de Deber/Ejercicio 2/Java/GestionProductos.java
import java.io.File;
import java.util.Scanner;
import java.io.FileNotFoundException;

public class GestionProductos {
    public static void main(String[] args) {
        // Ajusta la ruta si el archivo está en una subcarpeta
        File archivo = new File("productos.txt");
        
        double totalGeneral = 0;
        String productoCaro = "";
        double precioMaximo = -1;

        try (Scanner sc = new Scanner(archivo)) {
            System.out.println("-------------------------------------------------------");
            System.out.printf("%-15s %-10s %-10s %-10s%n", "PRODUCTO", "PRECIO", "CANT.", "SUBTOTAL");
            System.out.println("-------------------------------------------------------");

            while (sc.hasNext()) {
                String nombre = sc.next();
                double precio = sc.nextDouble();
                int cantidad = sc.nextInt();

                double subtotal = precio * cantidad;
                totalGeneral += subtotal;

                // Lógica para encontrar el producto más caro
                if (precio > precioMaximo) {
                    precioMaximo = precio;
                    productoCaro = nombre;
                }

                System.out.printf("%-15s %-10.2f %-10d %-10.2f%n", nombre, precio, cantidad, subtotal);
            }

            System.out.println("-------------------------------------------------------");
            System.out.printf("TOTAL GENERAL: %.2f%n", totalGeneral);
            System.out.println("PRODUCTO MAYOR (PRECIO): " + productoCaro);

        } catch (FileNotFoundException e) {
            System.out.println("Error: No se encontró el archivo 'productos.txt'.");
        }
    }
}